### ISSUE TITLE
[YSQL] -: Reliable way to wait for ysql startup

### ISSUE THREAD

**Issue by karishmalalani** [yugabyte/yugabyte-db#7574]
version:2.3.1.0

--problem--
ysqlsh refused the connection

bellow is example  
customer invokes yb-master/yb-tserver directly and waits until pg_isready returns true. See yugabyte-flags.txt for flags used.

they just tested yugabyted and it doesn't work very well also. Here's a minimal example to reproduce the race condition during startup:

---command-- 

yugabyte-2.3.1.0.hrt1/bin/yugabyted start --base_dir=/usr/tmp/vic/test --ycql_port 23001 --ysql_port 23002 --master_rpc_port 23003 --tserver_rpc_port 23004 && yugabyte-2.3.1.0.hrt1/bin/ysqlsh -h 127.0.0.1 -p 23002 -c '\dt+;'
```
+--------------------------------------------------------------------------------------------------+
|                                            yugabyted                                             |
+--------------------------------------------------------------------------------------------------+
| Status:                          Running                                                                    |
| Web console         : http://127.0.0.1:7000                                                      |
| JDBC                : jdbc:postgresql://127.0.0.1:23002/yugabyte?user=yugabyte&password=yugabyte |
| YSQL                : bin/ysqlsh  -p 23002                                                       |
| YCQL                : bin/ycqlsh 127.0.0.1 23001                                                 |
| Data Dir            : /usr/tmp/vic/test/data                                                     |
| Log Dir             : /usr/tmp/vic/test/logs                                                     |
| Universe UUID       : 16165789-d957-4c7b-9c56-99577d06b2e3                                       |
+--------------------------------------------------------------------------------------------------+

ysqlsh: could not connect to server: Connection refused
        Is the server running on host "127.0.0.1" and accepting
        TCP/IP connections on port 23002?
```

In other words, yugabyted start returns 0 doesn't always guarantee the database is ready to accept connections. If we add a sleep 1 between the two commands above, sqlsh will work as expected.
